web/services/getHolidays.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/*
+ * Copyright (C) 2021 Igalia, S.L. <info@igalia.com>
+ *
+ * This file is part of PhpReport.
+ *
+ * PhpReport is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PhpReport is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PhpReport.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use Phpreport\Web\services\HolidayService;
+
+define('PHPREPORT_ROOT', __DIR__ . '/../../');
+
+require_once(PHPREPORT_ROOT . "/vendor/autoload.php");
+require_once(PHPREPORT_ROOT . '/util/LoginManager.php');
+
+$init = $_GET['init'] ?? NULL;
+$end = $_GET['end'] ?? NULL;
+$sid = $_GET['sid'] ?? NULL;
+
+$loginManager = new \LoginManager();
+
+$holidayService = new HolidayService($loginManager);
+
+header('Content-Type: application/json; charset=utf-8');
+
+$holidays = $holidayService->getUserVacationsRanges($init, $end, $sid);
+
+if (array_key_exists('error', $holidays)) {
+    // user is logged out or doesn't have permission
+    http_response_code(403);
```

Comment:
Yep! I was thinking of creating a REST API tiny structure, that would handle all the requests and route internally to each service based on the URL, but I'm still structuring it in my head.. probably will send a PR for this in the near future :)

Reply:
Expanding a bit further:

So instead of having the inumerous getXXXX, updateXXX  files that we have, we would have only the TemplateService, HolidayService, TaskService... and so on classes that would have all the methods for that "domain" (template, task, project, user...) and then a single file to handle the requests that would route based on the url and call the respective class service.

This request handler would take care of parsing the request, returning xml or json, checking authentication and permissions, anything related to the HTTP request. All the business rules should be inside the Service classes.